src/share/export.js

```diff
@@ -0,0 +1,48 @@
+import fs from '../lib/fs';
+const sh = new fs.Shell();
+import WebTorrent from 'webtorrent';
+
+const files = [];
+
+document.getElementById('btnSeed').addEventListener('click', startSeeding);
+
+function processPath(path, next) {
+    if (path.endsWith('/')) {
+        return next();
+    }
+    fs.readFile(path, function(err, data) {
+        if (err) {
+            console.error(err);
+            return next(err);
+        }
+        data.name = path;
+        files.push(data);
+        next();
+    });
+}
+
+function startSeeding() {
+    document.getElementById('btnSeed').disabled = true;
+    let client = new WebTorrent();
+    sh.find('/', { exec: processPath }, err => {
+        if (err) {
+            console.error(err);
+            document.getElementById('magnetURI').innerHTML =
+                'Error seeding files';
+            return;
+        }
+        try {
```

Comment:
We don't need `try/catch` here, because `client.seed()` won't throw.  Instead, it will raise `error` events.  You need to add:

```js
client.on('error', function (err) { /* deal with errors here */})
```